kivy/tests/test_uix_layout.py

```diff
@@ -14,3 +15,99 @@ class UixLayoutTest(unittest.TestCase):
     def test_instantiation(self):
         with self.assertRaises(Exception):
             layout = Layout()
+
+
+@pytest.mark.parametrize(
+    "sh_min_vals, sh_max_vals, hint, expected_ratio", [
+        ((50, None, None), (None, None, None), (1, 1, 1), (2, 1, 1)),
+        ((None, 50, None), (None, None, None), (1, 1, 1), (1, 2, 1)),
+        ((None, None, 50), (None, None, None), (1, 1, 1), (1, 1, 2)),
+        ((20, None, None), (None, None, None), (1, 1, 1), (1, 1, 1)),
+        ((None, 20, None), (None, None, None), (1, 1, 1), (1, 1, 1)),
+        ((None, None, 20), (None, None, None), (1, 1, 1), (1, 1, 1)),
+        ((50, 50, None), (None, None, None), (1, 1, 1), (1, 1, 0)),
+        ((50, None, 50), (None, None, None), (1, 1, 1), (1, 0, 1)),
+        ((None, 50, 50), (None, None, None), (1, 1, 1), (0, 1, 1)),
+        ((20, 20, None), (None, None, None), (1, 1, 1), (1, 1, 1)),
+        ((20, None, 20), (None, None, None), (1, 1, 1), (1, 1, 1)),
+        ((None, 20, 20), (None, None, None), (1, 1, 1), (1, 1, 1)),
+        ((None, None, None), (50, None, None), (1, 1, 1), (1, 1, 1)),
+        ((None, None, None), (None, 50, None), (1, 1, 1), (1, 1, 1)),
+        ((None, None, None), (None, None, 50), (1, 1, 1), (1, 1, 1)),
+        ((None, None, None), (20, None, None), (1, 1, 1), (1, 2, 2)),
+        ((None, None, None), (None, 20, None), (1, 1, 1), (2, 1, 2)),
+        ((None, None, None), (None, None, 20), (1, 1, 1), (2, 2, 1)),
+        ((None, None, None), (50, 50, None), (1, 1, 1), (1, 1, 1)),
+        ((None, None, None), (50, None, 50), (1, 1, 1), (1, 1, 1)),
+        ((None, None, None), (None, 50, 50), (1, 1, 1), (1, 1, 1)),
+        ((None, None, None), (20, 20, None), (1, 1, 1), (1, 1, 3)),
+        ((None, None, None), (20, None, 20), (1, 1, 1), (1, 3, 1)),
+        ((None, None, None), (None, 20, 20), (1, 1, 1), (3, 1, 1)),
+        # a case where some space remaining.
+        ((None, None, None), (20, 20, 20), (1, 1, 1), (1, 1, 1)),
```

Comment:
I guess there is no way to tell remaining space (100 - 20 * 3) from the result of `layout_hint_with_bounds()`